lib/matplotlib/pyplot.py

```diff
@@ -2409,10 +2419,13 @@ def polar(*args, **kwargs):
     """
     # If an axis already exists, check if it has a polar projection
     if gcf().get_axes():
-        if not isinstance(gca(), PolarAxes):
+        ax = gca()
+        if isinstance(ax, PolarAxes):
+            return ax
+        else:
             _api.warn_external('Trying to create polar plot on an axis '
```

Comment:
I think this was wrong originally:
```suggestion
            _api.warn_external('Trying to create polar plot on an Axes '
```